tcf_website/management/commands/fix_multiple_objects_returned.py

```diff
@@ -0,0 +1,113 @@
+from functools import reduce
+from operator import or_
+
+from django.core.management.base import BaseCommand
+from django.db.models import Case, Count, F, FloatField, Q, Sum, Value, When
+
+from tcf_website.models import CourseInstructorGrade
+
+
+class Command(BaseCommand):
+    """This fixes existing `MultipleObjectsReturned` errors"""
+
+    help = 'Fix `MultipleObjectsReturned` errors that were caused by inconsistent email data'
+
+    def handle(self, *args, **options):
+        self.verbosity = options['verbosity']
+
+        # Number of cases we want to fix
+        qs = CourseInstructorGrade.objects\
+            .exclude(course=None)\
+            .values('course', 'first_name', 'last_name')\
+            .annotate(num=Count('id'))\
+            .filter(num__gt=1)
+        if self.verbosity > 0:
+            print(qs.count())
+
+        qs_without_num = qs.values('course', 'first_name', 'last_name')
+        total = CourseInstructorGrade.objects\
+            .filter(reduce(or_, [Q(**a) for a in qs_without_num]))\
+            .order_by('first_name', 'last_name', 'subdepartment', 'number')
+        without_email = total.filter(email='')
+
+        # Fix the weird case with two different emails for the same professor
+        to_fix = list(CourseInstructorGrade.objects.filter(
+            email='lr4zs@virginia.edu',
+        ))
+        for obj in to_fix:
+            obj.email = ''
+        CourseInstructorGrade.objects.bulk_update(to_fix, ['email'])
+
+        # Now everything should have been fixed.
+        if self.verbosity > 0:
+            print(without_email.count())
+
+        for a in qs_without_num:
+            instructors = list(
+                CourseInstructorGrade.objects.filter(**a).order_by('email')
+            )
+            # The second one has the email, and the first one doesn't

```

Comment:
Is this assuming that there are only two duplicates for any given instance? Is that a safe assumption?

Reply:
Clarified through comments in f45bd0586dca0e7cfc008ddf272e194f660bbc75 and added an assert statement